core/scripts/cli/sponsor/viewLiquidationDetails.js

```diff
@@ -1,24 +1,120 @@
+    )}, Status: ${liquidationStateToDisplay(liquidationState)}`;
+    choices.push({ name: display, value: liquidationId });
+  }
+  const input = await inquirer.prompt({
+    type: "list",
+    name: "choice",
+    message:
+      "Pick a liquidation. You can withdraw rewards from liquidations marked 'PENDING DISPUTE' or 'SUCCESSFULLY DISPUTED' if the oracle has resolved a price and that price makes the dispute successful.",
+    choices
+  });
+  if (input["choice"] === backChoice) {
+    return;
+  }
+
+  await viewWithdrawRewardsMenu(web3, artifacts, emp, liquidationStructs[input["choice"]], input["choice"]);
+};
+
+/**
+ * @notice Display menu of options to withdraw rewards from eligible liquidations.
+ */
+const viewWithdrawRewardsMenu = async (web3, artifacts, emp, liquidation, id) => {
+  // If liquidation `sponsor` property is empty, then either the sponsor's liquidation rewards have been withdrawn or the
+  // liquidation data has been deleted following an expired liquidation or a failed dispute.
+  // Therefore, this should catch all liquidations with state == `UNINITIALIZED` following a deletion of the liquidation struct.
+  if (liquidation.sponsor === "0x0000000000000000000000000000000000000000") {
+    console.log("There are no rewards to withdraw from this liquidation");
+    return;
+  }
 
+  // Now we know that the liquidation still has rewards to be withdrawn, but we need to check if the sponsor is eligible
+  // to receive rewards.
+  const ExpandedERC20 = artifacts.require("ExpandedERC20");
   const sponsorAddress = await getDefaultAccount(web3);
-  const display = `Liquidated at epoch time ${liquidation.liquidationTime} by ${liquidation.liquidator}`;
+  const liquidationTimeReadable = new Date(Number(liquidation.liquidationTime.toString()) * 1000);
+  const display = `Liquidated at time ${liquidationTimeReadable} by ${liquidation.liquidator}`;
   const backChoice = "Back";
-  const withdrawAction = "Withdraw";
+  // Sponsors can withdraw rewards only if a liquidation has been disputed successfully.
+  const withdrawAction = "Dispute succeeded; withdraw rewards";
   choices = [{ name: backChoice }];
+
   // Check if the sponsor can withdraw by seeing if `withdrawLiquidation` reverts.
   try {
     await emp.withdrawLiquidation.call(id, sponsorAddress);
     choices.push({ name: withdrawAction });
   } catch (err) {
-    // Withdraw wouldn't work so it shouldn't be a valid option.
+    // Withdraw wouldn't work so it shouldn't be a valid option. Let's print out a detailed response to the user to indicate why
+    // `withdraw` will fail.
+
+    // The sponsor can only withdraw rewards from a successfully disputed liquidation.
+    if (liquidation.state === LiquidationStatesEnum.PRE_DISPUTE) {
+      // If liquidation state is PRE_DISPUTE and `withdrawLiquidation` fails, then the liquidation has either expired
+      // or is pre-expiry. Only a liquidator can withdraw from an expired liquidation.
```

Comment:
Can't we differentiate between these two cases by looking at the expiration time on the liqudiation?

Reply:
Yup easily. Done PTAL!

Reply:
LGTM